sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -269,6 +273,62 @@ def update_leaves_values(self, grower, y_true, raw_predictions,
             # Note that the regularization is ignored here
 
 
+class Poisson(BaseLoss):
+    """Poisson deviance loss with log-link, for regression.
+
+    For a given sample x_i, Poisson deviance loss is defined as::
+
+        loss(x_i) = y_true_i * log(y_true_i/exp(raw_pred_i))
+                    - y_true_i + exp(raw_pred_i))
+
+    This actually computes half the Poisson deviance to simplify
+    the computation of the gradients.
+    """
+
+    def __init__(self, sample_weight):
+        # If sample weights are provided, the hessians and gradients
+        # are multiplied by sample_weight, which means the hessians are
+        # equal to sample weights.
```

Comment:
The comment was meant for e.g LS because normally LS has constant hessians (equal to 1), but they're not constant anymore when we have SW (where hessian = SW). But this doesn't apply for poisson since the hessians are never constant anyway, so let's remvoe the comment